### PR TITLE
Automatically use lucky egg for XP Boost.

### PR DESCRIPTION
**Pull request by xezero**
If specified in the config file, we can now have the bot automatically use lucky eggs every 30 minutes to double the XP :)

### CODE REVIEW

File: pgoapi/pgoapi.py [hunk 1]
```diff
@@ -224,6 +229,16 @@ def heartbeat(self):
             self.attempt_evolve(self.inventory.inventory_items)
             self.cleanup_pokemon(self.inventory.inventory_items)
 
+        # Auto-use lucky-egg if applicable
+        if self.auto_use_lucky_egg and self.inventory.has_lucky_egg() and current_time - self._last_egg_use_time > 30*60:
+            self.use_item_xp_boost(item_id=Inventory.ITEM_LUCKY_EGG)
+            response = self.call()
+
+            if 'USE_ITEM_XP_BOOST' in response['responses'] and 'result' in response['responses']['USE_ITEM_XP_BOOST'] and response['responses']['USE_ITEM_XP_BOOST']['result'] == 1:
```
Comment (joychugh) [inline pgoapi/pgoapi.py]: can we handle some other response codes here too? it would be good to see if something happened other than what we expected.

**Comment by joychugh** [comment]
👍  love it! thanks for the feature! just minor comment. in the PR

File: pgoapi/pgoapi.py (hunk 1 above)
Comment (xezero) [inline pgoapi/pgoapi.py]: Good question! I actually tried to see if I could get any other response code with other scenarios (for instance, if the lucky egg is already in use), and I don't even get the 'result' property back, hence why I have to check it in the conditional :(

Comment (joychugh) [inline pgoapi/pgoapi.py]: https://github.com/AeonLucid/POGOProtos/blob/master/src/POGOProtos/Networking/Responses/UseItemXpBoostResponse.proto check here. for response codes, you can see how we handle the response code in other areas 
like resp.get('result', -1)   resp being   the call response [responses][use_item_xp_boost]

**Comment by j-e-k** [comment]
Thanks for the contribution! 😄

File: pgoapi/pgoapi.py (hunk 1 above)
Comment (xezero) [inline pgoapi/pgoapi.py]: Thanks for polishing it up and accepting the PR! I don't know much of Python as I come from a different background so :)

**Pull request merged**